ziggurat-core-crawler/src/summary.rs

```diff
@@ -25,6 +34,8 @@ pub struct NetworkSummary {
     pub crawler_runtime: Duration,
     /// Addresses of good nodes.
     pub node_addrs: Vec<SocketAddr>,
+    /// Network types of good nodes. Indexes correspond to `node_addrs` and `node_indices`.
+    pub node_networks: Vec<NetworkType>,
     /// Unidirected connections graph.
     pub nodes_indices: NodesIndices,
```

Comment:
```
Indexes correspond to `node_addrs` and `node_indices`.
```

To remove this assumption - which the user of the NetworkSummary struct must be aware,
we can couple these there items into a single struct, so this comment can be completely avoided.
Something like:
```rust
struct NodeDescription {
    addr: SocketAddr
    network_type: NetworkType
    indices: Vec<usize>
}
```

The usage of the struct:
```rust
/// Contains stats about crawled network.
#[derive(Default, Clone, Deserialize, Serialize)]
pub struct NetworkSummary {
    /// Total number of nodes discovered.
    pub num_known_nodes: usize,
    /// Number of nodes that a crawler was able to connect to.
    pub num_good_nodes: usize,
    /// Total number of known connections as reported by peers.
    pub num_known_connections: usize,
    /// Number of all protocol versions discovered.
    pub num_versions: usize,
    /// Map: Version number -> number of nodes that reported this version.
    pub protocol_versions: HashMap<u32, usize>,
    /// Nodes' software versions.
    pub user_agents: HashMap<String, usize>,
    /// Crawler's runtime.
    pub crawler_runtime: Duration,
    /// Node list.
    pub nodes: Vec<NodeDescription>,
}
```

This way, all three node properties share the same index anytime the user accesses this vector field.

Reply:
> I can see a place to potential refactorization and create struct with Node which could collect information about IP, network and indices but for now the aim is to introduce new feat without bigger changes in zcash or xrpl crawlers.

I seem not to have read the PR description 😃 (feel free to resolve this one)